src/static/helpers/slas.ts

```diff
@@ -0,0 +1,277 @@
+
+  const crypto = await import("crypto");
+  const challenge = urlSafe(
+    crypto.default.createHash("sha256").update(codeVerifier).digest("base64")
+  );
+
+  if (challenge.length === 0) {
+    throw new Error("Problem generating code challenge");
+  }
+
+  return challenge;
+};
+
+/**
+ * Wrapper for the authorization endpoint. For federated login (3rd party IDP non-guest), the caller should redirect the user to the url in the url field of the returned object. The url will be the login page for the 3rd party IDP and the user will be sent to the redirectURI on success. Guest sessions return the code and usid directly with no need to redirect.
+ * @param slasClient - a configured instance of the ShopperLogin SDK client
+ * @param codeVerifier - random string created by client app to use as a secret in the request
+ * @param parameters - Request parameters used by the `authorizeCustomer` endpoint.
+ * @param parameters.redirectURI - the location the client will be returned to after successful login with 3rd party IDP. Must be registered in SLAS.
+ * @param parameters.hint? - optional string to hint at a particular IDP. Guest sessions are created by setting this to 'guest'
+ * @param parameters.usid? - optional saved SLAS user id to link the new session to a previous session
+ * @returns login url, user id and authorization code if available
+ */
+export async function authorize(
+  slasClient: ShopperLogin,
+  codeVerifier: string,
+  parameters: {
+    redirectURI: string;
+    hint?: string;
+    usid?: string;
+  }
+): Promise<{ code: string; url: string; usid: string }> {
+  const codeChallenge = await generateCodeChallenge(codeVerifier);
+
+  // Create a copy in order to modify behavior of the fetch call
+  // We do not want to redirect to redirectURI so manually control redirect
+  const slasClientCopy = new ShopperLogin(slasClient.clientConfig);
+  slasClientCopy.clientConfig.fetchOptions = {
+    ...slasClient.clientConfig.fetchOptions,
+    redirect: "manual",
+  };
+
+  const options = {
+    parameters: {
+      client_id: slasClient.clientConfig.parameters.clientId,
+      code_challenge: codeChallenge,
+      ...(parameters.hint && { hint: parameters.hint }),
+      organizationId: slasClient.clientConfig.parameters.organizationId,
+      redirect_uri: parameters.redirectURI,
+      response_type: "code",
+      ...(parameters.usid && { usid: parameters.usid }),
+    },
+  };
+
+  const response = await slasClientCopy.authorizeCustomer(options, true);
+  const redirectUrl = response.headers?.get("location") || response.url;
+
+  if (!redirectUrl) {
+    throw new Error("Authorization failed");
+  }
+
+  return { url: redirectUrl, ...getCodeAndUsidFromUrl(redirectUrl) };
+}
+
+/**
+ * A single function to execute the ShopperLogin Public Client Guest Login with proof key for code exchange flow as described in the [API documentation](https://developer.salesforce.com/docs/commerce/commerce-api/references?meta=shopper-login:Summary).
```

Comment:
Nit: If we're asking folks provide a client secret, then must use a SLAS _Private Client_ w/ PKCE.

Reply:
Hey @joeluong-sfcc, the comment of this function says:

> A single function to execute the ShopperLogin Public Client Guest Login with proof key for code exchange flow as described in the API documentation.

But the implementation is a private client flow using client credentials. _NOT a public client with PKCE_.

I'm saying the function comment should match what the function does.